pkg/webui/console/containers/gateway-managed-gateway/connection-settings/index.js

```diff
@@ -312,31 +329,43 @@ const GatewayConnectionSettings = () => {
 
       setError(undefined)
       try {
+        let body = {}
         const { wifi_profile, ethernet_profile } = values
 
-        const shouldUpdateNonSharedWifiProfile =
-          wifi_profile.profile_id === 'non-shared' &&
-          Boolean(nonSharedWifiProfileId) &&
-          wifi_profile._enable_wifi_connection
-
-        const wifiProfileId = await getWifiProfileId(wifi_profile, shouldUpdateNonSharedWifiProfile)
-        const ethernetProfileId = await getEthernetProfileId(
-          ethernet_profile,
-          cleanValues.ethernet_profile,
-        )
-        const body = {
-          wifi_profile_id:
-            !wifi_profile._enable_wifi_connection && !wifi_profile._override ? null : wifiProfileId,
-          ethernet_profile_id: ethernetProfileId,
+        let wifiProfileId
+        if (hasWifi) {
+          const shouldUpdateNonSharedWifiProfile =
+            wifi_profile.profile_id === 'non-shared' &&
+            Boolean(nonSharedWifiProfileId) &&
+            wifi_profile._enable_wifi_connection
+
+          wifiProfileId = await getWifiProfileId(wifi_profile, shouldUpdateNonSharedWifiProfile)
+          body = {
+            ...body,
+            wifi_profile_id:
+              !wifi_profile._enable_wifi_connection && !wifi_profile._override
+                ? null
+                : wifiProfileId,
+          }
```

Comment:
We can simplify the logic here to avoid sending unnecessary requests:

```suggestion
          if (!wifi_profile._enable_wifi_connection && !wifi_profile._override) {
            body.wifi_profile_id = null
          } else {
            body.wifi_profile_id = await getWifiProfileId(
              wifi_profile,
              shouldUpdateNonSharedWifiProfile,
            )
          }
```